chaincode/fabcar/go/contract/shim.go

```diff
@@ -0,0 +1,21 @@
+package contract
+
+import (
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+//go:generate counterfeiter -o mocks/transaction.go -fake-name TransactionContext . TransactionContext
```

Comment:
This is totally new way to unit test . Thanks

I am not familiar with counterfeiter. As a developer, it will be helpful for now, to add a section/README
(in  fabcar directory somewhere) with bullet list of steps to use counterfeiter & generating mocks

Reply:
Yes, as this was totally a spur of the moment thing without any input from the other maintainers, I decided to wait to add Doc until I gauged feedback and support. If the rest of the maintainers agree there is value in this, I plan to add a doc on unit testing in both Go and Node.

Reply:
I'll also mention, this is the correct way to unit test chaincode now as I deprecated the MockStub last year that is part of the `fabric-chaincode-go` project, with the expectation that chaincode developers would use generated mocks instead.

https://jira.hyperledger.org/browse/FAB-15367
https://github.com/hyperledger/fabric/commit/dac782a804cf796721ca0f5383b9a6a8a5d83b9f